google-ads-examples/src/main/java/com/google/ads/googleads/examples/reporting/ParallelReportDownload.java

```diff
@@ -0,0 +1,249 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.ads.googleads.examples.reporting;
+
+import com.beust.jcommander.Parameter;
+import com.google.ads.googleads.examples.utils.ArgumentNames;
+import com.google.ads.googleads.examples.utils.CodeSampleParams;
+import com.google.ads.googleads.lib.GoogleAdsClient;
+import com.google.ads.googleads.v3.errors.GoogleAdsError;
+import com.google.ads.googleads.v3.errors.GoogleAdsException;
+import com.google.ads.googleads.v3.services.GoogleAdsServiceClient;
+import com.google.ads.googleads.v3.services.SearchGoogleAdsStreamRequest;
+import com.google.ads.googleads.v3.services.SearchGoogleAdsStreamResponse;
+import com.google.api.gax.rpc.ResponseObserver;
+import com.google.api.gax.rpc.StreamController;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ParallelReportDownload {
+
+  // Adjust these fields as required.
+  /** Defines the Google Ads Query Language (GAQL) query strings to run for each customer ID. */
+  private static final List<String> GAQL_QUERY_STRINGS =
+      ImmutableList.of(
+          "SELECT campaign.id, metrics.impressions, metrics.clicks FROM campaign",
+          "SELECT campaign.id, ad_group.id, metrics.impressions, metrics.clicks FROM ad_group");
+
+  /** Specifies the maximum duration that this code example may run for. */
+  private static final int MAX_WAIT_TIME_SECONDS = 60 * 60; // 1 hour.
+
+  /** Absolute wall clock time by when the job must be completed. */
+  private static final long DEADLINE_MILLIS =
+      System.currentTimeMillis() + MAX_WAIT_TIME_SECONDS * 1000;
+
+  private static class ParallelReportDownloadParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    List<Long> customerIds;
+
+    @Parameter(names = ArgumentNames.LOGIN_CUSTOMER_ID)
+    Long loginCustomerId;
+  }
+
+  public static void main(String[] args) {
+    ParallelReportDownloadParams params = new ParallelReportDownloadParams();
+    if (!params.parseArguments(args)) {
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerIds = ImmutableList.of(Long.valueOf("INSERT CUSTOMER IDS"));
```

Comment:
Hmm, so I considered this but left it out to keep this example clearer - there's already a lot going on here and I didn't want to distract from the core message.

The `GetAccountHierarchy` example is really quite involved due to reconstructing the tree. Most of that logic isn't required here, just a simple list of customer IDs.

How about we add a separate example `ListCustomersUnderManager`? This would be short - it's just a call to select `customer_client` entries. Then we can reference that example in comments from here.

Reply:
Thought about this, and I see the benefits of keeping this example focused on parallel downloads instead of making it overly complicated and overly specialized for a specific use case for the source of the customer ID list.

I'm OK with keeping this as you have it, provided the class Javadoc here explicitly points the user to both the
[GetAccountHierarchy](https://github.com/googleads/google-ads-java/blob/master/google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/GetAccountHierarchy.java) 
and 
[ListAccessibleCustomers](https://github.com/googleads/google-ads-java/blob/master/google-ads-examples/src/main/java/com/google/ads/googleads/examples/accountmanagement/ListAccessibleCustomers.java)
examples as ways to retrieve a list of customer IDs for different use cases.

Reply:
sgtm - added this to the class javadoc.

Reply:
Looks good. Thanks for making those changes.